pages/common/highlight.md

```diff
@@ -0,0 +1,28 @@
+# highlight
+
+> Outputs syntax-highlighted source code to a variety of formats.
+> More information: <http://www.andre-simon.de/doku/highlight/en/highlight.php>.
+
+- Produce a complete HTML document from a source code file:
+
+`highlight -O html --style {{theme name}} --syntax={{language}} {{source file}}`
+
+- Produce an HTML fragment, suitable for inclusion in a larger document:
+
+`highlight -O html --fragment --syntax={{language}} {{source file}}`
+
+- Inline all CSS styling in every tag:
+
+`highlight -O html --inline-css --syntax={{language}} {{source file}}`
+
+- List all supported languages:
+
+`highlight --list-scripts lang`
```

Comment:
```suggestion
`highlight --list-scripts {{lang}}`
```

Reply:
The command to list supported languages is *exactly* `highlight --list-scripts lang`. In this case `lang` is a *keyword*, not a variable.

Reply:
But `lang` is one of multiple arguments for `--list-scripts`. We use token syntax for this too. And it's obvious from the description, that you don't have to put in a specific language.

Reply:
`{{lang}}` syntax implies that the character sequence "lang" must be substituted for an actual value. However, in this case, the character sequence "lang" **is** the value.

The `--list-scripts` accepts multiple *fixed* keywords. In the BNF it could be written `--list-scripts <"lang"|"themes"|"plugins">`. I'll reiterate: "lang" (sic), "themes", and "plugins" are keywords, and they are the only keywords that the `--list-scripts` option accepts.

```
$ highlight --list-scripts lang | head

Installed langDefs (located in /usr/share/highlight/langDefs/):

ABAP/4                        : abap4 ( abp )
ABC                           : abc
Advanced Backus-Naur Form     : abnf
ActionScript                  : actionscript ( as )
ADA95                         : ada ( a adb ads gnad )
Agda                          : agda
ALAN Interactive Fiction Language: alan ( alan i )
```

Reply:
But that's also what we use token syntax for. `{{lang}}` doesn't necessarily have to substituted with anything. Everything, which isn't a flag or subcommand goes into token syntax.

Reply:
Please read my previous comment carefully. The command to list all installed language syntaxes (that is, Lua scripts of the type `lang`) is **exactly** `highlight --list-scripts lang`. Writing it `--list-scripts {{lang}}` would be the same as writing `git {{checkout}}`—that would be invalid syntax.

That command is to be copied/pasted or typed unmodified. This is what will happen if the user copies `highlight --list-scripts {{lang}}` and runs it:

```
$ highlight --list-scripts {{lang}}
highlight: Unknown script type '{{lang}}'. Apply one of 'themes', 'langs' or 'plug-ins'.
```

Because the "lang" word in `--list-scripts lang` not only does not need to be substituted with anything, but **MUST NOT** be substituted with anything if your goal is to list scripts of the type "lang", using a meta-variable syntax for it is unnecessary and confusing. 

Here's another example for you: `ping` supports `-M` option whose arguments can be **exactly** `do`, `dont`, or `want`. 

```
       -M pmtudisc_opt
           Select Path MTU Discovery strategy.  pmtudisc_option may be either do (prohibit fragmentation, even local one), want (do PMTU discovery, fragment locally when packet size is large), or dont (do not set DF flag).
```

By your logic the command for sending ICMP echo requests with the DF (Don't Fragment) bit set must be written `ping -M{{do}} {{host}}`. The syntax that does make sense would be `ping -Mdo {{host}}`—the `{{host}}` part must be substituted with an actual host address, but `-Mdo` is exactly the option the user wants to disable fragmentation.

**OR** we would write `ping -M<do|dont|want> {{host}}` to show the user all available options.

Reply:
I changed the example `--list-scripts {{langs|themes|plugins}}` and reworded its description accordingly.

Reply:
Something worthy of discussion is that it there is ambiguity here though, to be honest.

I do think it could make sense to consider if there is a way to denote that the parameter is intended literally, but I don't know a good way to do that.

Reply:
* The `{{ }}` won't be rendered it's just there to hightlight that part. Like this:
![image](https://user-images.githubusercontent.com/50295997/136213408-6f8483d4-45ba-4d9e-9091-4777176ef419.png)

* I don't think the `|` will get confused with a pipe, because no one writes a pipe without spaces between it.

Reply:
Oh, I see. Somehow I never made a connection between curly braces and the output in the client, sorry!